### PR TITLE
Store the age restriction that will be applied on HyprMX init.

### PR DESCRIPTION
**Pull request by cb-jpadilla**
- To keep behavior similar to our iOS.

### CODE REVIEW

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 1]
```diff
@@ -298,6 +295,31 @@ class HyprMXAdapter : PartnerAdapter {
...
+     *
+     * @param context a context that will be passed to the SharedPreferences to get the age restriction.
+     *
+     * @return An already stored age restriction.
+     */
+    private fun getAgeRestricted(context: Context) =
+        context.getSharedPreferences(
+        HYPRMX_PREFS_KEY,
+        Context.MODE_PRIVATE
+    ).getBoolean(HYPRMX_AGE_RESTRICTION_KEY, false)
```
Comment (chauduyphanvu) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:309]: Nit: indentation/formatting looks off

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 2]
```diff
@@ -298,6 +295,31 @@ class HyprMXAdapter : PartnerAdapter {
...
+     * Store a HyprMX age restriction value and set it to HyprMX.
+     * This is passed on SDK initialization.
+     *
+     * @param context a context that will be passed to the SharedPreferences to set the user consent.
+     * @param isAgeRestricted the age restriction value to be stored.
+     */
+    private fun setAgeRestricted(context: Context, isAgeRestricted: Boolean) {
+        context.getSharedPreferences(HYPRMX_PREFS_KEY, Context.MODE_PRIVATE).edit()
+            .putBoolean(HYPRMX_AGE_RESTRICTION_KEY, isAgeRestricted).apply()
+    }
```
Comment (chauduyphanvu) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:321]: Nit: formatting for more readability:

```kt
    private fun setAgeRestricted(context: Context, isAgeRestricted: Boolean) {
        context.getSharedPreferences(HYPRMX_PREFS_KEY, Context.MODE_PRIVATE)
            .edit()
            .putBoolean(HYPRMX_AGE_RESTRICTION_KEY, isAgeRestricted)
            .apply()
    }
```

**Review by chauduyphanvu — APPROVED** [review]
lgtm once formatting is taken care of

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt (hunk 2 above)
Comment (cb-jpadilla) [reply HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:321]: Will make it consistent. 👍🏼

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 3]
```diff
@@ -298,6 +298,33 @@ class HyprMXAdapter : PartnerAdapter {
...
+     */
+    private fun getAgeRestricted(context: Context) =
+        context.getSharedPreferences(
+            HYPRMX_PREFS_KEY,
+            Context.MODE_PRIVATE
+        ).getBoolean(HYPRMX_AGE_RESTRICTION_KEY, false)
+
+    /**
+     * Store a HyprMX age restriction value and set it to HyprMX.
+     * This is passed on SDK initialization.
```
Comment (bichenwang) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:316]: What is passed on sdk init? Coppa? It's definitely set in set age restricted, not init. 
This method is called in init? That is not useful documentation for this method. This method could be called from anywhere that finds it useful.
Or do you mean coppa usually is set on sdk init like from mediation?

**Review by bichenwang — APPROVED** [review]
Looks good. Just comments

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt (hunk 3 above)
Comment (cb-jpadilla) [reply HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:316]: Oh, yeah, let me rephrase this. Thanks for pointing it out. 👍🏼

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 4]
```diff
@@ -264,8 +261,11 @@ class HyprMXAdapter : PartnerAdapter {
...
     private fun setUserConsent(context: Context, consentStatus: ConsentStatus) {
-        context.getSharedPreferences(HYPRMX_PREFS_KEY, Context.MODE_PRIVATE).edit()
-            .putInt(HYPRMX_USER_CONSENT_KEY, consentStatus.ordinal).apply()
+        context.getSharedPreferences(HYPRMX_PREFS_KEY, Context.MODE_PRIVATE)
+            .edit()
+            .putInt(HYPRMX_USER_CONSENT_KEY, consentStatus.ordinal)
+            .apply()
+
         HyprMX.setConsentStatus(consentStatus)
     }
```
Comment (chauduyphanvu) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:270]: not something immediately but to be safe here and elsewhere I'd recommend `.commit()` over `.apply()` since we will know the result of the write before we decide if we should call `HyprMX.setConsentStatus(consentStatus)`. otherwise like right now the two things will just happen independently of each other—you could end up with a failed SharedPreferences write but still call `HyprMX.setConsentStatus(consentStatus)`. and when you get the value back from the SP later it doesn't exist so it'll be confusing.

something like this but you can simplify it:

```kt
private fun setUserConsent(context: Context, consentStatus: ConsentStatus) {
    val prefsWriteSucceeded = context.getSharedPreferences(HYPRMX_PREFS_KEY, Context.MODE_PRIVATE)
        .edit()
        .putInt(HYPRMX_USER_CONSENT_KEY, consentStatus.ordinal)
        .commit()

    when(prefsWriteSucceeded) {
        true -> {
            HyprMX.setConsentStatus(consentStatus)
        }
        else -> {
            // Log or handle the error
        }
    }
}
```

but, that's a really edge case 😛

**Review by chauduyphanvu — APPROVED** [review]

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt (hunk 4 above)
Comment (cb-jpadilla) [reply HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:270]: Sure. Let's aim to be safer. 🗒️ 👍🏼

Comment (bichenwang) [reply HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:270]: doesn't strict mode flag for using commit on the main thread? It's probably fine since it's one int. either way, it's probably fine.
oh, you're in a suspend function already. you could make this a suspend function and boop this over to the IO thread and then commit() is fine.

Comment (cb-jpadilla) [reply HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt]: Oh, yeah, I can turn them to a suspend function and do the commit() there. Let me do that. 🗒️ ✍🏼

Comment (cb-jpadilla) [reply HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt]: @bichenwang actually, should we still do the suspend thing:
image

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 5]
```diff
@@ -261,12 +268,22 @@ class HyprMXAdapter : PartnerAdapter {
...
+
+        PartnerLogController.log(
+            CUSTOM,
+            "User consent ${
+                if (prefsWriteSucceeded) "was" else "was not"
+            } successfully stored."
+        )
 
-        HyprMX.setConsentStatus(consentStatus)
+        when (prefsWriteSucceeded) {
```
Comment (bichenwang) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:282]: I'm not sure using a `when` for a boolean is necessary
```
if (prefsWriteSucceeded) {
    HyprMX.setConsentStatus(consentStatus)
}
```
seems sufficient

**Review by bichenwang — APPROVED** [review]
you can do the commit on the main thread or whatever. it's a small difference. if you want to, you can bump it on the background thread with withContext since you're already in a coroutine, but it's honestly not a big deal
lgtm otherwise
